web-components/src/components/input/Input.ts

```diff
@@ -614,10 +614,13 @@ export namespace Input {
     }
 
     private characterCountLabelTemplate() {
-      return this.maxLength && this.displayCharacterCount && !this.disabled && !this.readOnly
```

Comment:
![09_33_36](https://github.com/user-attachments/assets/3ed604f3-3658-47ed-b973-b0802b1b9d14)

Tweaking this again, the ask is to have this label work more as a "suggested" limit rather than a hard limit. So the user can still enter more characters, but we show an error message beneath the box if they do.